### DISCUSSION TITLE
Handling events and atomic UI update

### DISCUSSION THREAD

**Discussion by phuoccss** [statelyai/xstate#1302] [General] [answered]
In Redux, multiple reducers can handle the same dispatched action. And I can connect different parts of the React UI to different reducers so that re-rendering is more atomic.

Just wondering if there is similar concept for XState, whereby multiple machines/services can respond to the same event, instead of having to call `send` for each machine services.

Currently I can think of 2 approaches:

1. Parallel states. Problem is I can only connect the UI to the whole machine service, not to each parallel state atomically.

2. Child machines that will handle the same event. However, I am not sure how to connect my UI to each individual child machine services.

Would love to hear your thoughts on this.

**Comment by Andarist** [comment, marked as answer]
> Just wondering if there is similar concept for XState, whereby multiple machines/services can respond to the same event, instead of having to call send for each machine services.

You can use `autoForward`, but it is strongly advisable that you actually forward events between machines manually. You can also share machines addresses (their sessions ids) so you can target events more directly, rather than having to pass events through all the layers each time.
